fs/src/jvmMain/kotlin/com/suparnatural/core/fs/FileSystem.kt

```diff
@@ -0,0 +1,228 @@
+                canonicalPath = PathComponent(file.canonicalPath),
+                createdAt = 0.0,
+                modifiedAt = file.lastModified().toDouble(),
+                size = file.length().toDouble(),
+                type = fileType
+        )
+    }
+
+    actual fun readDir(path: String): List<StatResult>? {
+        val parent = File(path).canonicalFile ?: return null
+        val fileList = parent.listFiles() ?: return null
+        return fileList.map {
+            buildStats(it)
+        }
+    }
+
+
+    actual fun readDir(pathComponent: PathComponent): List<StatResult>? {
+        val path = pathComponent.component ?: return null
+        return readDir(path)
+    }
+
+
+    actual fun stat(path: String): StatResult? {
+        val file = File(path)
+        return buildStats(file)
+    }
+
+
+    actual fun stat(pathComponent: PathComponent): StatResult? {
+        val path = pathComponent.component ?: return null
+        return stat(path)
+    }
+
+
+    actual fun readFile(path: String, encoding: ContentEncoding): String? {
+        val file = File(path).canonicalFile
+
+        val charset = {
+            when (encoding) {
+                ContentEncoding.Ascii -> Charsets.US_ASCII
+                else -> Charsets.UTF_8
+            }
+        }()
+        val reader = BufferedReader(InputStreamReader(FileInputStream(file), charset))
+        val content = reader.readLines().joinToString("\n")
+
+        if (encoding == ContentEncoding.Base64) {
+            return String(Base64.getDecoder().decode(content), Charsets.UTF_8)
+        }
+        return content
+    }
+
+
+    actual fun readFile(pathComponent: PathComponent, encoding: ContentEncoding): String? {
+        val path = pathComponent.component ?: return null
+        return readFile(path, encoding)
+    }
+
+    private fun writeFile(path: String, contents: String, create: Boolean, encoding: ContentEncoding, append: Boolean = false): Boolean {
+        val file = File(path).canonicalFile
+        if (!file.exists()) {
+            if (!create) return false
+            file.createNewFile()
+        }
+
+        val finalContent: String
+        val appendToFile: Boolean
+
+        if (encoding == ContentEncoding.Base64) {
+            val sourceString = if (append) (readFile(path, ContentEncoding.Base64) ?: "") + contents else contents
+
+            finalContent = Base64.getEncoder().encodeToString(sourceString.toByteArray(Charsets.UTF_8))
+            appendToFile = false
+
+        } else {
+            finalContent = contents
+            appendToFile = append
+        }
+
+        val charset = {
+            when (encoding) {
+                ContentEncoding.Ascii -> Charsets.US_ASCII
+                else -> Charsets.UTF_8
+            }
+        }()
+        val bufferedWriter = BufferedWriter(OutputStreamWriter(FileOutputStream(file, appendToFile), charset))
+        bufferedWriter.write(finalContent)
```

Comment:
There should be some IOException handling. If writer throw exception, stream will not be closed and will leak.

E.g. bufferedWriter.use {
bufferedWriter.write(finalContent)
} //will close writer automatically and ignore exception (if that was intended)